### DISCUSSION TITLE
Re-order autocomplete options based on likelihood of use

### DISCUSSION THREAD

**Discussion by ZencareJake** [zed-industries/zed#21222] [Feature Requests]
Occasionally, options in autocomplete menu seem arbitrarily ordered. For instance, in the screenshot below, `filteredTagString` is what I want to be suggested first in the autocomplete menu. Instead, it is the 6th option shown. I'm not sure if there settings I could adjust that might satisfy this request, or if adjusting the order of suggested options would require going under the hood. In any case, any insight would be much appreciated!

<img width="984" alt="Screenshot 2024-11-26 at 11 54 01 AM" src="https://github.com/user-attachments/assets/06532a2c-8800-44ba-9ef3-337793f8cd2e">

**Comment by hmbakhsh** [comment]
I've run into very similar situations whereby in VSCode I'd get intellisense/autocomplete in (a) better order (b) much shorter/refined list of completions

![image](https://github.com/user-attachments/assets/80c20d9b-b7df-4961-8e54-d4ba95d2904d "VS Code")
**VS Code: Typing in 1 character - it's the 4th option**

![image](https://github.com/user-attachments/assets/dadd3794-0cfd-4c4f-8d81-7d615874aeb7)
**Zed: Typing in 4 characters - it's the 9th option**

**Reply by lungarella-raffaele** [reply]
Sorry out of topic question. It seems like in the zed editor you have different colors for different kind of completions for types/variables, how can I accomplish this? I really miss this feature in vscode where it gives more info about the intellisense

**Comment by T1ckbase** [comment]
Same here, `document` should be the first option.

<img width="999" height="391" alt="image" src="https://github.com/user-attachments/assets/c762b8d2-71ca-411f-b287-8a50599b965c" />

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
